Added expressions to conditionals and renamed conditional to conditions

ggetz opened the pull request:
Changed `conditional` to `conditions` and added `expression` property.

Also added new tests to test `expression`.

pjcozzi commented on Specs/Scene/ConditionalExpressionSpec.js:
```diff
@@ -24,13 +24,22 @@ defineSuite([
         }
     };
 
+    var jsonExpWithExpression = {
```
Looks good.  There should also be a test to verify that `${expression} === undefined` when `expression` is not there, and a check to verify that `evaluate` returns undefined when `conditions` is not there.  These are some of the corners in the schema that I alluded to in https://github.com/AnalyticalGraphicsInc/3d-tiles/pull/80

ggetz commented:
I added more tests to cover the cases you mentioned, and in doing so, realized there was no case in expression for a literal `undefined`, so I added it.

pjcozzi commented:
:+1:
